### PR TITLE
registered user can log in and out

### PR DESCRIPTION
**Pull request by damwhit**
a registered user has the ability to log in and out of an existing account

### CODE REVIEW

File: spec/features/user/user_can_login_and_out_of_existing_accounts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    fill_in "password", with: user.password
+
+    click_on "login"
+
+    within("div .alert") do
+      expect(page).to have_content("Hi John, welcome to ticketcacher!")
+    end
+
+    expect(current_path).to eq("/users/#{user.id}/dashboard")
+
```
Comment (erinnachen) [inline spec/features/user/user_can_login_and_out_of_existing_accounts_spec.rb:29]: could we make this route  '/dashboard' ?

File: spec/features/user/user_can_login_and_out_of_existing_accounts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    expect(page).to_not have_link("login")
+    expect(page).to have_content("John")
+    expect(page).to have_link("logout")
+    expect(page).to have_link("order history")
+
+    click_on "logout"
+
```
Comment (erinnachen) [inline spec/features/user/user_can_login_and_out_of_existing_accounts_spec.rb:43]: Is there a flash message that should be displayed when one logs out?

**Comment by damwhit** [comment]
asserted logout flash in test and changed user dashboard path to be '/dashboard'

**Comment by damwhit** [comment]
added sad paths

**Comment by cheljoh** [comment]
Looking good Davey, thanks for adding sad paths!

**Pull request merged**